### DISCUSSION TITLE
Flashing to release from old snapshot

### DISCUSSION THREAD

**Discussion by zembtach** [dangowrt/owrt-ubi-installer#132] [Q&A] [answered]
Hi y'all. Sorry if this is kind of similar to #121 but I need to ask for help. 😅

I've left my RT3200 on a really old release as I wasn't using it until now. I think it was on v0.6.0 from this repo, or something. Possibly older. However, I did just now run the _Attended Sysupgrade_ from the web interface and that seemed to work fine!

I am on the UBI layout already. Can I just flash [`openwrt-22.03.2-mediatek-mt7622-linksys_e8450-ubi-squashfs-sysupgrade.itb`](https://downloads.openwrt.org/releases/22.03.2/targets/mediatek/mt7622/openwrt-22.03.2-mediatek-mt7622-linksys_e8450-ubi-squashfs-sysupgrade.itb) right onto it to have everything updated to the release version?

I'm curious about the U-Boot version and ARM trusted firmware and stuff like that. Will that be upgraded automatically as well? I'm not familiar at all with how everything is put together for a targeted device release. This is my first time using OpenWRT. 🙏

I would just try and see what happens, but I would prefer to not brick the device if it can be avoided! 😅

**Comment by dangowrt** [comment, marked as answer]
> I've left my RT3200 on a really old release as I wasn't using it until now. I think it was on v0.6.0 from this repo, or something. Possibly older.

If you are currently running anything from [v0.5.3](https://github.com/dangowrt/owrt-ubi-installer/releases/tag/v0.5.3) (ie. openwrt/openwrt@050621a) on, then yes, you can just upgrade to newer releases. You can even use `auc` on the command line to do so automatically:
```
auc -b 22.03 -B 22.03.2
```
should do the trick. In this way, all additionally installed packages (like the automated update UI in LuCI, `auc` on the command line, ...) will again be installed.

If the device is currently running anything *older than* [v0.5.3](https://github.com/dangowrt/owrt-ubi-installer/releases/tag/v0.5.3) (ie. openwrt/openwrt@050621a) then you will need to flash the most recent version of the installer first. Do **not** do this without good reason, see below.

> However, I did just now run the Attended Sysupgrade from the web interface and that seemed to work fine!

Ok, if that already worked and you are on newest snapshot now and 5 GHz WiFi is still working, then you are fine. If (and *only* if) you are having problems with either of the WiFi interface while seeing errors reporting read-errors from MTD in the kernel log you are advised to re-run the installed.

> I am on the UBI layout already. Can I just flash [`openwrt-22.03.2-mediatek-mt7622-linksys_e8450-ubi-squashfs-sysupgrade.itb`](https://downloads.openwrt.org/releases/22.03.2/targets/mediatek/mt7622/openwrt-22.03.2-mediatek-mt7622-linksys_e8450-ubi-squashfs-sysupgrade.itb) right onto it to have everything updated to the release version?

Now that you have already run the attended sysupgrade in LuCI and made sure that WiFi is still working fine, yes, you can do that.

> I'm curious about the U-Boot version and ARM trusted firmware and stuff like that. Will that be upgraded automatically as well?

No, only OpenWrt will be updated. Updating TF-A and U-Boot is too risky to be carried out automatically and/or without a well-defined need to do so. Also note that if your device was already working fine, nothing will be improved at this point by updating TF-A or U-Boot.

> I'm not familiar at all with how everything is put together for a targeted device release. This is my first time using OpenWRT. pray
> 
> I would just try and see what happens, but I would prefer to not brick the device if it can be avoided! sweat_smile

The replacement bootloader is designed to make it very hard to brick the device, so don't worry too much. Even if something goes wrong (let's say: you ignore all warnings and flash a wrong firmware file) in most cases the device will just come up with the recovery firmware which allows you to comfortably load working firmware onto the device. In rare cases flashing a wrong firmware can destroy the recovery OS as well, but even then the device will request a new recovery firmware via TFTP and then boot that.

So unless you are messing with the `read-only` partitions, you'll always be fine.
